### PR TITLE
Autogenerate gcb yamls

### PR DESCRIPTION
**Pull request by alex1545**
Automatically generate a GCB yaml file for every GCB build.
This eliminates the maintenance of different versions of GCB yaml files and gets rid of the substitution logic inside the yaml file.

### CODE REVIEW

File: container/build.py [hunk 1]
```diff
@@ -259,16 +262,26 @@ def cloud_build(project,
       full_path = os.path.join(dirpath, f)
       os.chmod(full_path, 0o644)
 
+  # create the required steps for the GCB yaml file as a list of dicts
+  gcb_steps = get_gcb_steps(project, container, tag, package, target,
```
Comment (xingao267) [inline container/build.py]: I'm not sure about have these additional set of flags to choose whether or not add this or that step. We could list out which steps are needed to be added directly. Sort of moving and unpacking get_gcb_steps() here.

File: container/build.py [hunk 2]
```diff
@@ -259,16 +262,26 @@ def cloud_build(project,
...
-    # Gets the yaml relative to the bazel-toolchains root, regardless of what directory it was called from
-    # MUST BE UPDATED IF THE YAML FILE IS MOVED
-    config_file = "{}/container/cloudbuild_no_bucket.yaml".format(
-        bazel_toolchains_base_dir)
-    extra_substitution = ""
+  # MUST BE UPDATED IF THE YAML FILE IS GENERATED ELSEWHERE
+  config_file = "{}/container/cloudbuild-{}.yaml".format(bazel_toolchains_base_dir, type_)
+
+  # create the GCB yaml file
+  gen_gcb_yaml_file(gcb_yaml_dict, config_file)
```
Comment (xingao267) [inline container/build.py]: Let's add some basic tests that validate that a gcb yaml file generated match some expected content.

File: container/gcb_yaml_gen_utils.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    List of steps (dicts), where each step is created by the
+    create_step function.
+  """
+  # indicate to use the same name for all steps if needed
+  if len(name_list) == 1:
+  	name_list *= len(args_list)
+
+  steps_list = []
+
+  # create all the steps one at a time
```
[the remark this answers is not in the document]
Comment (alex1545) [reply container/gcb_yaml_gen_utils.py]: I ran pyformat to fix all such issues.

File: container/build.py (hunk 2 above)
Comment (alex1545) [reply container/build.py]: I added unit tests under test/container.

**Comment by xingao267** [comment]
Closing this for now as the plan has changed.

**Pull request closed without merging**